Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -456,8 +456,10 @@ TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)
 		}
 	}
 
-	// If at least one entry was not used for the same frame, overwrite the oldest one
-	if (temp_frameCount != 0x7fffffff)
+	// If at least one entry was not used for the same frame, overwrite the oldest one. Some games reuse paletted
+	// textures after long periods of time, so don't overwrite old cache entries when loading paletted textures
+	// Example: Sonic the Fighters(inside Sonic Gems Collection) reuses paletted textures up to 64 frames later
+	if (temp_frameCount != 0x7fffffff && !isPaletteTexture)
```

Comment:
Well, the 64 frames animation uses a texture with the same texture offset, the same texture hash, and different palettes. Dolphin uses 32 as palette size, and the min/max calculation results in min=0 and max=15, so the actual size of those palettes is 32.